### DISCUSSION TITLE
Devnet Pre-Release 1.4.1alpha1

### DISCUSSION THREAD

**Discussion by amc-ie** [MinaProtocol/mina#14943] [Announcements]
The O(1) Labs team is proud to announce that 1.4.1alpha1 is ready to test on Devnet! This release includes a set of important bugfixes, improvements as well as code cleanup.

If you run into any issues, please include `Release: 1.4.1alpha1` in the name of any issues you create. Please report any issues you encounter [here](https://github.com/MinaProtocol/mina/issues/new/choose) or join the [Discussion]()!

#### Bug Fixes

- Disable snark pool persistence and additional daemon performance fixes due to snark pool https://github.com/MinaProtocol/mina/pull/13409
- Fix for long async cycles impacting block production performance https://github.com/MinaProtocol/mina/pull/13654
- Fix for when a slot to start replaying from doesn’t have a block https://github.com/MinaProtocol/mina/pull/13765
- Replayer app fixes: https://github.com/MinaProtocol/mina/pull/13406 https://github.com/MinaProtocol/mina/issues/13401 https://github.com/MinaProtocol/mina/pull/13600

#### Improvements

- Add a GraphQL query to generate a config file for a hard fork https://github.com/MinaProtocol/mina/pull/13787
- Support larger ledgers in IPC communication https://github.com/MinaProtocol/mina/pull/11973
- Remove memory stats https://github.com/MinaProtocol/mina/pull/12904
- Remove support for bionic and stretch https://github.com/MinaProtocol/mina/pull/13657

### Complete Changelog since 1.4.0:
https://github.com/MinaProtocol/mina/compare/1.4.0...1.4.1alpha1

### Debian Packages:
**Debian Repository:**
First, set up and update the `alpha` Debian Repository for your platform with the following commands:
```
sudo rm /etc/apt/sources.list.d/mina*.list
echo "deb [trusted=yes] http://packages.o1test.net $(lsb_release -cs) alpha" | sudo tee /etc/apt/sources.list.d/mina-alpha.list
sudo apt-get update
```
Then, install the package(s) that you need:
**Mina Daemon:** `sudo apt-get install -y mina-devnet=1.4.1alpha1-3f42da6`
**Mina Archive:** `sudo apt-get install -y mina-archive=1.4.1alpha1-3f42da6`
**New Mina Archive Schema:** [create_schema.sql](https://github.com/MinaProtocol/mina/blob/da12bcf65fc6b37d81b237e57ab9284270d339ef/src/app/archive/create_schema.sql)

### Docker Images:
Every docker image is built for each of the supported platforms, replace `CODENAME` in the images below with your choice of `focal`,  `buster`, or `bullseye` to select the base docker image.

**Mina Daemon:** `minaprotocol/mina-daemon:1.4.1alpha1-3f42da6-CODENAME-devnet`
**Mina Archive:** `minaprotocol/mina-archive:1.4.1alpha1-3f42da6-CODENAME`

### Sandbox Node: For testing in an isolated, single-node network without snarks
As of 1.2.0, the demo functionality is built into the standard daemon container, via the environment variable `RUN_DEMO=true`.
Just `docker run -it --name mina-demo -e RUN_DEMO=true minaprotocol/mina-daemon:1.4.1alpha1-3f42da6-CODENAME-devnet` and go!

### Step by Step Guide:
Check out [our documentation](https://docs.minaprotocol.com/node-operators/connecting-to-devnet) for complete instructions on using this version to connect to Devnet. 
*Docker only* Make sure to run with `--peer-list-url https://storage.googleapis.com/seed-lists/devnet_seeds.txt`. or add `PEER_LIST_URL=https://storage.googleapis.com/seed-lists/devnet_seeds.txt` to your `.mina-env`.

If you are running the correct version on the correct network **DEVNET**, `mina client status` will show:
```
Chain id:     b6ee40d336f4cc3f33c1cc04dee7618eb8e556664c2b2d82ad4676b512a82418
Git SHA-1:    3f42da6683ed6bc17d00ff3158e2e68ad9df11c3
```

<hr /><em>This discussion was created from the release <a href='https://github.com/MinaProtocol/mina/releases/tag/1.4.1alpha1'>Devnet Pre-Release 1.4.1alpha1</a>.</em>

**Comment by jrwashburn** [comment]
While removing support for bionic and stretch, it would be very helpful to add support for bookworm.

**Reply by amc-ie** [reply]
@jrwashburn thank you, this is currently on our backlog and we hope to get to it soon! :)
